prowler/providers/azure/services/keyvault/keyvault_service.py

```diff
@@ -116,6 +121,25 @@ def __get_secrets__(self, subscription, resource_group, keyvault_name):
             )
         return secrets
 
+    def __get_vault_monitor_settings__(self, audit_info, keyvault_name, resource_group):
+        logger.info(
+            f"KeyVault - Getting monitor diagnostics settings for {keyvault_name}..."
+        )
+        monitor_diagnostics_settings = []
+        for subscription in self.clients:
+            try:
+                for client in monitor_client.clients.values():
+                    monitor_diagnostics_settings = monitor_client.diagnostic_settings_with_uri(
+                        subscription,
+                        f"subscriptions/{self.subscriptions[subscription]}/resourceGroups/{resource_group}/providers/Microsoft.KeyVault/vaults/{keyvault_name}",
+                        client,
+                    )
```

Comment:
```suggestion
            try:
                    monitor_diagnostics_settings = monitor_client.diagnostic_settings_with_uri(
f"subscriptions/{self.subscriptions[subscription]}/resourceGroups/{resource_group}/providers/Microsoft.KeyVault/vaults/{keyvault_name}",
                        monitor_client.clients[subscription],
                    )
```

Reply:
If im not iterating by subscriptions `self.subscriptions[subscription]` won't work because KeyVault object has no attribute `subscription`